python/domain_decomposition.cpp

```diff
@@ -72,12 +106,13 @@ void register_domain_decomposition(pybind11::module& m) {
     // Partition load balancer
     // The Python recipe has to be shimmed for passing to the function that takes a C++ recipe.
     m.def("partition_load_balance",
-        [](std::shared_ptr<py_recipe>& recipe, const context_shim& ctx) {
-            return arb::partition_load_balance(py_recipe_shim(recipe), ctx.context);
+        [](std::shared_ptr<py_recipe>& recipe, const context_shim& ctx, std::unordered_map<arb::cell_kind, arb::partition_hint>& hint_map) {
```

Comment:
This can be simplified a bit, by passing the `hint_map` by value, using the convenience type `arb::partition_hint_map`, and using the C++ type as the default value (no need for pybind11 to convert a Python dictionary to a C++ hash table).

```C++
    m.def("partition_load_balance",
        [](std::shared_ptr<py_recipe>& recipe,
           const context_shim& ctx,
           arb::partition_hint_map hint_map)
        {                                                                                                
            return arb::partition_load_balance(py_recipe_shim(recipe),
                                               ctx.context,
                                               std::move(hint_map));
        },
        "Construct a domain_decomposition that distributes the cells in the model described by recipe\n"
        "over the distributed and local hardware resources described by context.\n"
        "Optionally, provide a dictionary of partition hints for certain cell kinds, by default empty.",
        "recipe"_a, "context"_a, "hints"_a=arb::partition_hint_map{});
```

Reply:
Done.